.github/workflows/test.yml

```diff
@@ -45,7 +47,8 @@ jobs:
           {
```

Comment:
@MichaelOrlov you should be able to manually set the build & test packages through `colcon-defaults`, e.g.,

```
{
  "build": {
    "packages-up-to": [
      "pkg_a",
      "pkg_b"
    ]
  },
  "test": {
    "packages-select": [
      "pkg_a"
    ]
  }
}
```

but you need to remove the `package-name:` input

Reply:
@christophebedard I have tried 
```
colcon-defaults: |
{
  rosbag2_path=$(colcon list -p --packages-select rosbag2)/..
  rosbag2_packages=$(colcon list -n --base-paths ${rosbag2_path})
  "build": {
    "cmake-args": [
      "-DCMAKE_CXX_FLAGS=\"-Werror\"",
      "-DBUILD_ROSBAG2_BENCHMARKS=1"
    ]
  },
  "test": {
    "ctest-args": ["-LE", "xfail"],
    "pytest-args": ["-m", "not xfail"],
    "packages-select": [${rosbag2_packages}],
    "packages-skip": [rosbag2_performance_benchmarking]
  }
}
```
and even moving 
```
  rosbag2_path=$(colcon list -p --packages-select rosbag2)/..
  rosbag2_packages=$(colcon list -n --base-paths ${rosbag2_path})
```
out of `{..}` scope operator.
Although still getting **Error: colcon-defaults value is not a valid JSON string:** in CI build_and_test run.

Don't know how to properly provide those list without `rosbag2_performance_benchmarking` package.
I will revert my original change in `test.yaml` file to be able proceed for code review.

Reply:
@MichaelOrlov like I said here https://github.com/ros2/rosbag2/pull/1268/files#r1152156032, you cannot use commands. This is not a script. This is a string. As I said here https://github.com/ros2/rosbag2/pull/1268/files#r1148171130, you must explicitly write the names of the packages just like the `package-name:` input.

So:

```
    - name: Build and run tests
      id: action-ros-ci
      uses: ros-tooling/action-ros-ci@v0.2
      with:
        target-ros2-distro: rolling
        vcs-repo-file-url: https://raw.githubusercontent.com/ros2/ros2/master/ros2.repos
        colcon-defaults: |
          {
            "build": {
              "cmake-args": [
                "-DCMAKE_CXX_FLAGS=\"-Werror\"",
                "-DBUILD_ROSBAG2_BENCHMARKS=1"
              ],
              "packages-up-to": [
                "mcap_vendor",
                "ros2bag",
                "rosbag2",
                "rosbag2_compression",
                "rosbag2_compression_zstd",
                "rosbag2_cpp",
                "rosbag2_examples_cpp",
                "rosbag2_interfaces",
                "rosbag2_performance_benchmarking_msgs",
                "rosbag2_performance_benchmarking",
                "rosbag2_py",
                "rosbag2_storage",
                "rosbag2_storage_mcap",
                "rosbag2_storage_mcap_testdata",
                "rosbag2_storage_sqlite3",
                "rosbag2_test_common",
                "rosbag2_tests",
                "rosbag2_transport",
                "shared_queues_vendor",
                "sqlite3_vendor",
                "zstd_vendor"
              ]
            },
            "test": {
              "ctest-args": ["-LE", "xfail"],
              "pytest-args": ["-m", "not xfail"],
              "packages-select": [
                "mcap_vendor",
                "ros2bag",
                "rosbag2",
                "rosbag2_compression",
                "rosbag2_compression_zstd",
                "rosbag2_cpp",
                "rosbag2_examples_cpp",
                "rosbag2_interfaces",
                "rosbag2_performance_benchmarking_msgs",
                "rosbag2_py",
                "rosbag2_storage",
                "rosbag2_storage_mcap",
                "rosbag2_storage_mcap_testdata",
                "rosbag2_storage_sqlite3",
                "rosbag2_test_common",
                "rosbag2_tests",
                "rosbag2_transport",
                "shared_queues_vendor",
                "sqlite3_vendor",
                "zstd_vendor"
              ]
            }
          }
```

Reply:
@christophebedard Thank, I've got it. Giving it a try. Will see if it will works.

Reply:
It seems it works.